### PR TITLE
chore(electric): revert removal of electric_user_id filtering "hack"

### PR DESCRIPTION
**Pull request by magnetised**

### CODE REVIEW

**Comment by alco** [comment]
@magnetised It would be nice to learn about the rationale for this change.

File: components/electric/lib/electric/features.ex [hunk 1]
```diff
@@ -90,6 +90,14 @@ defmodule Electric.Features do
     :ok
   end
 
+  def list(name \\ __MODULE__) do
+    name
+    |> :ets.match(:"$1")
+    |> Enum.into(%{}, fn [f] -> f end)
+    |> Map.merge(Process.get({__MODULE__, name}, %{}))
+    |> Enum.to_list()
```
Comment (alco) [inline components/electric/lib/electric/features.ex:98]: If the intent here is to get all table entries, I find `:ets.tab2list()` to be more descriptive. And the temporary conversion to map is not strictly necessary here:

```suggestion
    features = :ets.tab2list(name)
    proc_overrides = Process.get({__MODULE__, name}, %{})
    
    Enum.reduce(proc_overrides, features, fn {key, val}, features ->
      Keyword.put(features, key, val)
    end)
```

File: components/electric/lib/electric/replication/changes/ownership.ex [hunk 2]
```diff
@@ -0,0 +1,47 @@
+defmodule Electric.Replication.Changes.Ownership do
+  alias Electric.Replication.Changes
+  alias Electric.Satellite.Auth
+
+  @user_id_column "electric_user_id"
+
+  @spec id_column_name() :: Changes.db_identifier()
+  def id_column_name, do: @user_id_column
```
Comment (alco) [inline components/electric/lib/electric/replication/changes/ownership.ex:8]: I'd use the same name for the function here, i.e. `user_id_column`. I had to look this implementation up to learn what `id_column_name()` meant when it was referenced in another module.

File: components/electric/lib/electric/replication/changes/ownership.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  @doc """
+  Given a change, checks if belongs to the given user_id.
+
+  For a change to belong to a user with id `user_id` the row must satisfy one of the conditions:
+
+  1. The table being modified has an `#{@user_id_column}` and the value of this column either
```
Comment (alco) [inline components/electric/lib/electric/replication/changes/ownership.ex:15]:
```suggestion
  1. The table being modified has an `#{@user_id_column}` column and the value of this column either
```

File: components/electric/lib/electric/replication/changes/ownership.ex [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  def change_belongs_to_user?(%Changes.Migration{}, _user_id, _owner_column) do
+    true
+  end
+
+  @empty [nil, ""]
+
+  @spec validate_record(Changes.record(), Auth.user_id(), Changes.db_identifier()) :: boolean()
+  defp validate_record(record, user_id, owner_column) do
+    case record do
```
Comment (alco) [inline components/electric/lib/electric/replication/changes/ownership.ex:40]: It's a matter of taste, I guess, but `case Map.get(record, owner_column) do` would yield less repetition below.

**Review by alco — APPROVED** [review]

File: components/electric/lib/electric/replication/changes/ownership.ex (hunk 3 above)
Comment (magnetised) [reply components/electric/lib/electric/replication/changes/ownership.ex:15]: This ownership file is just being un-deleted, and has a limited lifespan, so though I appreciate your feedback, I'm just going to leave as-is.

File: components/electric/lib/electric/features.ex (hunk 1 above)
Comment (icehaunter) [reply components/electric/lib/electric/features.ex:98]: Or you could do a list conversion which is also super simple

```suggestion
    overrides =
      {__MODULE__, name}
      |> Process.get(%{})
      |> Map.to_list()

    name
    |> :ets.tab2list()
    |> Keyword.merge(overrides)
```

Comment (magnetised) [reply components/electric/lib/electric/features.ex:98]: Thanks for the reminder re tab2list. I could re-implement this as you say but I personally don't see any great advantage to your alternative versions considering the test-only role of this function.

**Pull request merged**